sdk/keyvault/azure-security-keyvault-keys/src/list_keys_single_page_result.cpp

```diff
@@ -125,3 +127,32 @@ DeletedKeySinglePage _detail::KeyPropertiesSinglePageSerializer::DeletedKeySingl
 
   return deletedKeySinglePage;
 }
+
+void DeletedKeySinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  GetDeletedKeysSinglePageOptions options;
+  options.NextPageToken = NextPageToken;
+  *this = keyClient->GetDeletedKeysSinglePage(options, context);
+  CurrentPageToken = NextPageToken.Value();
+}
+
+void KeyPropertiesSinglePage::OnNextPage(const Azure::Core::Context& context)
+{
+  // Before calling `OnNextPage` pagedResponse validates there is a next page, so we are sure
+  // NextPageToken is valid.
+  if (m_type == KeyPropertiesSinglePage::KeyPropertiesType::Keys)
+  {
+    GetPropertiesOfKeysSinglePageOptions options;
+    options.NextPageToken = NextPageToken;
+    *this = m_keyClient->GetPropertiesOfKeysSinglePage(options, context);
+  }
+  else if (m_type == KeyPropertiesSinglePage::KeyPropertiesType::Versions)
+  {
+    GetPropertiesOfKeyVersionsSinglePageOptions options;
+    options.NextPageToken = NextPageToken;
+    *this = m_keyClient->GetPropertiesOfKeyVersionsSinglePage(m_keyName, options, context);
+  }
+  CurrentPageToken = NextPageToken.Value();
```

Comment:
So `CurrentPageToken` always has the same value as `NextPageToken`? This doesn't look right.

Reply:
My bad, I should've used the options value for the previous NextPageToken.  Thank you